packages/yew-macro/tests/props_macro/props-pass.rs

```diff
@@ -56,6 +56,15 @@ pub struct RawIdentProps {
     r#pointless_raw_name: ::std::primitive::usize,
 }
 
+#[derive(::std::cmp::PartialEq)]
+pub struct Stub<T>(::std::marker::PhantomData<::std::boxed::Box<T>>);
+
+#[derive(::yew::Properties, ::std::cmp::PartialEq)]
+pub struct SelfRefProps<T: ::std::cmp::PartialEq> {
+    x: Stub<T>,
+    y: Stub<Self>,
```

Comment:
```suggestion
    x: Box<T>,
    y: Box<Self>,
```

Using an existing `T: ?Sized` type should already suffice.

Reply:
Could you please also add a couple additional test cases?

```rust
z: &'a Self,
a: PhantomData<(&'a Self, Self)>,
b: PhantomData<Box<Self>>,
c: fn(&Self) -> Self,
```